astropy/time/core.py

```diff
@@ -483,6 +485,12 @@ class TimeBase(ShapedLikeNDArray):
     def __getnewargs__(self):
         return (self._time,)
 
+    def __getstate__(self):
+        # For pickling, we remove the cache from what's pickled
+        state = (self.__dict__ if PYTHON_LT_3_11 else super().__getstate__()).copy()
+        state.pop("_id_cache", None)
```

Comment:
This is necessary because `WeakValueDictionary` cannot be pickled (the similar change in `TimeFormat` is just because there is no sense to pickle a cache generally).